lib/sdd/framework/utils.bash

```diff
@@ -420,18 +419,30 @@ _tag_name_of_latest_github_release() {
     # Fetch tag name of latest release on GitHub
     local github_user=$1
     local repo_name=$2
-    wget -qO- https://api.github.com/repos/"$github_user"/"$repo_name"/releases/latest | grep tag_name | awk '{ print $2; }' | sed 's/[",]//g'
+
+    # Get tag name from URL redirection to avoid GitHub API limits
+    awk -F'[ /]' -v e=1 '/Location:/ {e=0; print $(NF-1);} END {exit(e);}' < <(
+        wget -o- --max-redirect 0 "https://github.com/$github_user/$repo_name/releases/latest"
+    )
 }
 
 _sha_of_github_master() {
     # Fetch SHA of latest commit on GitHub master branch
     local github_user=$1
     local repo_name=$2
-    wget -qO- https://api.github.com/repos/"$github_user"/"$repo_name"/commits/master | grep -m1 sha | awk '{ print $2; }' | sed 's/[",]//g'
+
+    # Get latest commit hash from atom feed to avoid GitHub API limits
+    wget -qO- "https://github.com/$github_user/$repo_name/commits/master.atom" |
+        awk -F'[/<>]' '/<id>tag:github.com,2008:Grit::Commit/ {i[n++] = $(NF-3);}
+        END {print i[0];}'
```

Comment:
For personal understanding: we search for lines containing `/<id>tag:github.com,2008:Grit::Commit/` and in those use `/<>` as field separators. The third field (why take the total number of fields `NF` into account?) is stored in an array `i`. Eventually we print the first element.
Could we also return right after we find the first match?

Reply:
> why take the total number of fields NF into account?

It gets the 3rd field from the end of a line, just because it was easier to count for me at that moment.

> Could we also return right after we find the first match?

Yes, if you do not plan using `set -o pipefail`, otherwise wget will return exit code 3 (File I/O error) because awk closes pipe before wget ends downloading the XML.

Reply:
Is using `set -o pipefail` generally recommended?
Instead of `NF-3`, can we use `4`? I find it a bit clearer.

Reply:
@pylipp 

> Is using set -o pipefail generally recommended?

`set -euo pipefail` is generally recommended. I using the options in my fork, but it too hard to implement in this project. I didn't find the easy way to use `set -e` with sdd_{install,uninstall,upgrade} functions.

> Instead of NF-3, can we use 4? I find it a bit clearer.

Course you can change it if you want.

Reply:
Thx for the insights!